app/src/main/java/com/asfoundation/wallet/ui/iab/AppcoinsRewardsBuyInteract.kt

```diff
@@ -0,0 +1,26 @@
+package com.asfoundation.wallet.ui.iab
+
+import com.appcoins.wallet.bdsbilling.WalletService
+import com.asfoundation.wallet.support.SupportInteractor
+import io.reactivex.Completable
+import io.reactivex.Single
+
+class AppcoinsRewardsBuyInteract(private val inAppPurchaseInteractor: InAppPurchaseInteractor,
+                                 private val supportInteractor: SupportInteractor,
+                                 private val walletService: WalletService) {
+
+  fun showSupport(gamificationLevel: Int): Completable {
+    return walletService.getWalletAddress()
+        .flatMapCompletable {
+          Completable.fromAction {
+            supportInteractor.registerUser(gamificationLevel, it.toLowerCase())
```

Comment:
we should have it.lowercase using the implicit locale as default. Use Locale.ROOT instead